### PR TITLE
DISPATCH-1513 - Fixed router crash when running http requests over ht…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…tps. This fix is available only on systems that are running libwebsockets version 3.2.0 and above

### CODE REVIEW

File: tests/system_tests_http.py.in [hunk 1]
```diff
@@ -250,8 +250,9 @@ def listener(**kwargs):
         self.assert_get("https://localhost:%s" % r.ports[0])
         # requireSsl=false Allows simple-ssl HTTP
 
-        # Commenting out the following assert until DISPATCH-1513 is fixed.
-        #self.assert_get("http://localhost:%s" % r.ports[0])
+        # The following test will be commented out if you are runnung a version lower than
```
Comment (kgiusti) [inline tests/system_tests_http.py.in]: "runnung"

File: src/http-libwebsockets.c [hunk 2]
```diff
@@ -354,7 +354,11 @@ static void listener_start(qd_http_listener_t *hl, qd_http_server_t *hs) {
 
         info.options |=
             LWS_SERVER_OPTION_DO_SSL_GLOBAL_INIT |
+#if (LWS_LIBRARY_VERSION_MAJOR == 3 && LWS_LIBRARY_VERSION_MINOR >= 2)
```
Comment (kgiusti) [inline src/http-libwebsockets.c]: What happens when LWS_LIBRARY_VERSION_MAJOR == 4?

File: tests/system_tests_http.py.in (hunk 1 above)
Comment (ChugR) [reply tests/system_tests_http.py.in]: These are compile-time tests that build software with or without some features. Don't we really want code that is compiled to handle both cases and then detect what to do at run-time?

Comment (ganeshmurthy) [reply tests/system_tests_http.py.in]: We do have c code that is conditionally compiled using an extra flag based on the version of LWS available on the system. The python tests have to be dealt with separately as well, so we test the version in the  cmake/FindLibWebSockets.cmake  file

**Pull request closed without merging**